amethyst_renderer/src/vertex.rs

```diff
@@ -3,48 +3,66 @@
 use std::any::Any;
 
 use gfx;
-use gfx::format::Format;
-use gfx::pso::buffer::Structure;
+use gfx::format::{ChannelType, Format, SurfaceType};
+use gfx::pso::buffer::Element;
 use gfx::traits::Pod;
 
-/// Handle to a vertex attribute.
-pub type Attribute = gfx::pso::buffer::Element<Format>;
+pub trait Attribute {
+    const NAME: &'static str;
+    const FORMAT: Format;
+    const SIZE: u32; // Has to be equal to `std::mem::size_of::<Self::Repr>() as u32`
```

Comment:
Make comment a doc comment?

Reply:
And in general, please add some docs.

Reply:
I forgot to stage docs before commiting 😄